src/app/service/scoap-squad/ScoapDatabase.ts

```diff
@@ -0,0 +1,106 @@
+import dbInstance from '../../utils/db';
+import { Db } from 'mongodb';
+import cloneDeep from 'lodash.clonedeep';
+import { ScoapEmbed, VoteRecord } from './ScoapClasses';
+import client from '../../app';
+import { TextChannel } from 'discord.js';
+import channelIds from '../constants/channelIds';
+import ScoapUtils from '../../utils/ScoapUtils';
+import constants from '../constants/constants';
+import { createReactionCollector, collectReactions } from './ScoapPoll';
+
+// ScoapSquad state initialization
+export const scoapEmbedState = {};
+export const botConvoState = {};
+export const voteRecordState = {};
+ScoapUtils.logToFile(`state objects initiated. \n scoapEmbedState: ${JSON.stringify(scoapEmbedState)} \n botConvoState: ${JSON.stringify(botConvoState)}  \n voteRecordState: ${JSON.stringify(voteRecordState)}`);
+setInterval(function() { ScoapUtils.purgeExpiredBotConvo(botConvoState); }, 60000);
+
+
+const decircularizeScoapEmbed = (scoapEmbed) => {
+	const clone = cloneDeep(scoapEmbed);
+	clone.setCurrentMessage(scoapEmbed.getCurrentMessage().id);
+	clone.setCurrentChannel(scoapEmbed.getCurrentChannel().id);
+	return clone;
+};
+
+export const updateScoapEmbedAndVoteRecordDb = async (scoapEmbed, voteRecord) => {
+	const clone = decircularizeScoapEmbed(scoapEmbed);
+	const filter = { id: scoapEmbed.getId() };
+	const options = { upsert: true };
+	const updateDoc = { $set: { scoapEmbed: clone, voteRecord: voteRecord } };
+	const db: Db = await dbInstance.dbConnect('degen');
+	const dbScoap = db.collection('scoapSquad');
```

Comment:
Can this collection string 'scoapSquad' get added to constants file? Also what is the schema for this collection? We've been adding the datastructure for each collection under `/schema`

Reply:
It is basically a object dump. I initially tried to restore it by replacing the prototype of the restored object with the classes prototype but ran into some issues with some methods not working so I changed how the object is restored. The way it is now I can simplify the schema and define it.

Reply:
Ok so no scehma needed got it. Can we still update the constants.ts with
```
DB_COLLECTION_SCOAP_SQUARD: 'scoapSquad'
```

That way all of the data collection names that the bot interacts with can easily be seen in that section of code. That'll be important even when the collection hasn't been created or is deleted.

Reply:
yep done